ui/src/containers/BeamlineSetupContainer.jsx

```diff
@@ -45,7 +43,7 @@ function BeamlineSetupContainer(props) {
             value={motor.value}
             min={motor.limits[0]}
             max={motor.limits[1]}
-            step={step}
```

Comment:
I've hit a bit of a snag here.

The `diffractometer.beamstop_distance` has no corresponding `uiproperties` object (which is where the step is stored). If I add one by declaring a _Beamstop distance_ component in `ui.yaml`, under `beamline_setup`, then it shows up as a beamline attribute in the UI (next to _Energy_, _Resolution_, etc.)

Any thoughts?

Reply:
Hm, I don't have a good solution for it right away, one way would be to add property called display that can be either true or false and that would simply be displayed in the beamline setup. Id prefer another solution though because that is rather counter intuitive.

Perhaps by adding the possibility to have sub-components, which will make things rather complicated. Would it make sense to add another section called `beamstop` like:
```
beamstop:
  id: beamstop
  components:
    - label: Beamstop distance
      attribute: diffractometer.beamstop_distance
      role: beamstop_distance
      step: 0.01
      precision: 2
      suffix: mm
```

Reply:
Right, not a bad idea! :thinking: 

I can't help but think that the UI config needs a bigger overhaul, though. Is it okay if we keep the hardcoded step for now and if I add add a link to this thread in a comment so we can revisit it later?

Reply:
Yes, your probably right, the idea was to keep it very simple. But it of-course need to solve our main problems :)